pkgs/test_core/lib/src/runner/dart2js_compiler_pool.dart

```diff
@@ -93,19 +93,24 @@ class Dart2JsCompilerPool extends CompilerPool {
     });
   }
 
-  // TODO(nweiz): Remove this when sdk#17544 is fixed.
   /// Fix up the source map at [mapPath] so that it points to absolute file:
   /// URIs that are resolvable by the browser.
   void _fixSourceMap(String mapPath) {
-    var map = jsonDecode(File(mapPath).readAsStringSync());
+    var map = jsonDecode(File(mapPath).readAsStringSync()) as Map;
     var root = map['sourceRoot'] as String;
 
-    map['sources'] = map['sources'].map((Object? source) {
-      var url = Uri.parse('$root$source');
-      if (url.scheme != '' && url.scheme != 'file') return source;
-      if (url.path.endsWith('/runInBrowser.dart')) return '';
-      return p.toUri(mapPath).resolveUri(url).toString();
-    }).toList();
+    final mapUri = p.toUri(mapPath);
+    map.update(
+      'sources',
+      (sources) => [
+        for (var source in sources as List<Object?>)
```

Comment:
nit: I think I would prefer to just call `cast` on the map as opposed to doing the individual casts here

Reply:
A cast on the map would not do what we need, we can only cast the generic `V` to  `Object?`, not `List<Object?>` like we need.

Reply:
I mean a literal call to `.cast<String, List<Object?>>()`, which will have the same effect (lazily casting each value as its pulled out), but it encodes that assumption a bit earlier and more directly imo.

Ultimately, I am fine with either.

Reply:
WDYT about the latest commit?

```dart
    map.cast<String, List<Object?>>().update(
          'sources',
          (sources) => [
            for (var source in sources)
```